vault/external_tests/identity/login_mfa_okta_test.go

```diff
@@ -181,72 +202,62 @@ path "secret/foo" {
 }
 
 func TestInteg_LoginMFAOkta(t *testing.T) {
-	t.Skip("This test requires manual intervention and OKTA verify on cellphone is needed")
-	cluster := minimal.NewTestSoloCluster(t, nil)
-	client := cluster.Cores[0].Client
-
-	// Enable Userpass authentication
-	err := client.Sys().EnableAuthWithOptions("userpass", &api.EnableAuthOptions{
-		Type: "userpass",
-	})
-	if err != nil {
-		t.Fatalf("failed to enable userpass auth: %v", err)
+	if os.Getenv(logicaltest.TestEnvVar) == "" {
+		t.Skip("This test requires manual intervention and OKTA verify on cellphone is needed")
 	}
 
-	err = mfaGenerateOktaLoginMFATest(client)
-	if err != nil {
-		t.Fatalf("Okta failed: %s", err)
+	// Ensure each cred is populated.
+	credNames := []string{
+		"OKTA_ORG",
+		"OKTA_API_TOKEN",
+		"OKTA_USERNAME",
 	}
-}
+	testhelpers.SkipUnlessEnvVarsSet(t, credNames)
 
-func mfaGenerateOktaLoginMFATest(client *api.Client) error {
-	var err error
+	cluster := minimal.NewTestSoloCluster(t, nil)
+	client := cluster.Cores[0].Client
 
-	auths, err := client.Sys().ListAuth()
-	if err != nil {
-		return fmt.Errorf("failed to list auth mounts")
-	}
-	mountAccessor := auths["userpass/"].Accessor
+	// Enable Userpass authentication
+	mountAccessor := testhelpers.SetupUserpassMountAccessor(t, client)
 
-	_, err = client.Logical().Write("auth/userpass/users/testuser", map[string]interface{}{
-		"password": "testpassword",
-	})
-	if err != nil {
-		return fmt.Errorf("failed to configure userpass backend: %v", err)
-	}
+	// Create testuser entity and alias
+	entityClient, entityID, _ := testhelpers.CreateCustomEntityAndAliasWithinMount(t,
+		client, mountAccessor, "userpass", "testuser",
+		map[string]interface{}{
+			"name": "test-entity",
+			"metadata": map[string]string{
+				"email": os.Getenv("OKTA_USERNAME"),
+			},
+		})
 
-	secret, err := client.Logical().Write("identity/entity", map[string]interface{}{
-		"name": "test-entity",
-		"metadata": map[string]string{
-			"email": "<okta username>",
-		},
-	})
+	err := mfaGenerateOktaLoginMFATest(t, entityClient, mountAccessor, entityID)
 	if err != nil {
-		return fmt.Errorf("failed to create an entity")
+		t.Fatalf("Okta failed: %s", err)
 	}
-	entityID := secret.Data["id"].(string)
+}
 
-	_, err = client.Logical().Write("identity/entity-alias", map[string]interface{}{
-		"name":           "testuser",
-		"canonical_id":   entityID,
-		"mount_accessor": mountAccessor,
-	})
-	if err != nil {
-		return fmt.Errorf("failed to create an entity alias")
-	}
+func mfaGenerateOktaLoginMFATest(t *testing.T, client *api.Client, mountAccessor, entityID string) error {
+	t.Helper()
 
 	var methodID string
 	var userpassToken string
+
+	// OKTA_USE_TOTP allows the test runner to decide whether to use TOTP or Push verification.
+	useTOTP := os.Getenv("OKTA_USE_TOTP") != ""
```

Comment:
Setting `OKTA_USE_TOTP=1` switches the test from testing with Okta Push to testing with Okta TOTP.